### ISSUE TITLE
Enhancement: should throw if invalid `tsconfigRootDir`  is provided

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#11460]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I expect that these should throw (when validating parser options):

```ts
import tseslint from 'typescript-eslint';

export default tseslint.config(
  tseslint.configs.strictTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: 'not/an/absolute/path',
      }
    }
  }
);
```
(or `tsconfigRootDir: 42`, or `tsconfigRootDir: '.'`).

In other words, I think the following...

https://github.com/typescript-eslint/typescript-eslint/blob/7126c3a970d606ae36000a5ca8039373ce2df8fc/packages/typescript-estree/src/parseSettings/createParseSettings.ts#L47-L56

...should have (roughly)

```ts
assert(
  tsestreeOptions.tsconfigRootDir == null ||
    (typeof tsestreeOptions.tsconfigRootDir === "string" &&
      path.isAbsolute(tsestreeOptions.tsconfigRootDir)),
);
```

### Additional Info

Relates to https://github.com/typescript-eslint/typescript-eslint/issues/11414, https://github.com/typescript-eslint/typescript-eslint/issues/11429

**Comment by kirkwaiblinger** [comment]
If someone _really_ wants to have `tsconfigRootDir` be the current `cwd` (for some strange reason), `tsconfigRootDir: process.cwd()` ought to work.

**Comment by JoshuaKGoldberg** [comment]
Agreed. Throwing an error on an _**invalid**_ path is a Good Thing ™️ that will expose legitimate issues for configs that were probably not working as expected in the first place.

Emphasis added in case someone drives by later with protests around breaking changes. 😇
